pinot-broker/src/main/java/com/linkedin/pinot/broker/requesthandler/BrokerRequestHandler.java

```diff
@@ -511,14 +512,17 @@ private BrokerResponse processOptimizedBrokerRequests(@Nonnull BrokerRequest ori
     int numServersResponded = 0;
     long deserializationStartTime = System.nanoTime();
     Map<ServerInstance, DataTable> dataTableMap = new HashMap<>();
+    // Add a mutable long to sum the response sizes.
+    MutableLong totalResponseSize = new MutableLong(0);
     if (offlineServerResponseMap != null) {
       numServersResponded += offlineServerResponseMap.size();
-      deserializeServerResponses(offlineServerResponseMap, true, dataTableMap, offlineTableName, processingExceptions);
+      deserializeServerResponses(offlineServerResponseMap, true, dataTableMap, offlineTableName,
+          processingExceptions, totalResponseSize);
     }
     if (realtimeServerResponseMap != null) {
       numServersResponded += realtimeServerResponseMap.size();
       deserializeServerResponses(realtimeServerResponseMap, false, dataTableMap, realtimeTableName,
-          processingExceptions);
+          processingExceptions, totalResponseSize);
```

Comment:
Seems like you have to pass this argument because computation and emitting of metric is happening in different methods. Why can't deserializeServerResponse do the computation and directly emit the metric?

Reply:
That's because the response may come from both realtime and offline servers. If we make `deserializeServerResponses` do the computation and emit the metric, the metric only represents either realtime or offline total response size. Thus, I'm thinking whether we sum the totalResponseSize and then emit it in one time.